### PR TITLE
Kotlin + AGP + Gradle + Dependency updates + Cleanup

### PR DESCRIPTION
**Pull request by findjigar**

### CODE REVIEW

File: extensions/kermit-koin/build.gradle.kts
Comment (psh) [inline extensions/kermit-koin/build.gradle.kts:47]: With all the other cleanup, perhaps this might be the time to delete the commented out targets?

File: samples/sample-production/app/src/main/java/co/touchlab/kermitSample/FirstFragment.kt
Comment (psh) [inline samples/sample-production/app/src/main/java/co/touchlab/kermitSample/FirstFragment.kt:11]: package names should be all lower case

File: extensions/kermit-koin/build.gradle.kts
Comment (findjigar) [reply extensions/kermit-koin/build.gradle.kts:47]: I thought it's there to avoid confusion for reader around why we don't have those targets in `kermit-koin` while being there in `kermit` in general. So I kept it

File: samples/sample-production/app/src/main/java/co/touchlab/kermitSample/FirstFragment.kt
Comment (findjigar) [reply samples/sample-production/app/src/main/java/co/touchlab/kermitSample/FirstFragment.kt:11]: https://kotlinlang.org/docs/coding-conventions.html#naming-rules

> you do need to use multiple words, you can either just concatenate them together or use the camel case (org.example.myProject).

Also, IDE doesn't complain anymore with `kermitSample` but it was complaining with `KermitSample`.

Let me know if you think, as a TL practice, we should have package name entirely `lower case`. I'll make the change then.

File: gradle/libs.versions.toml
Comment (kpgalligan) [inline gradle/libs.versions.toml:22]: Should be 2.0.2. Had to fix deployment.

File: gradle/libs.versions.toml
Comment (kpgalligan) [inline gradle/libs.versions.toml:25]: Also whatever the new release is, probably 0.8.3

**Review by kpgalligan — APPROVED** [review]
Besides the 2 version updates, this looks good

File: gradle/libs.versions.toml
Comment (findjigar) [reply gradle/libs.versions.toml:25]: will wait for CrashKiOS release before updating and merging this PR. Thanks

**Pull request merged**
